commons-math/src/userguide/java/org/apache/commons/math4/userguide/genetics/HelloWorldExample.java

```diff
@@ -0,0 +1,187 @@
+
+        // Get the end time for the simulation.
+        long endTime = System.currentTimeMillis();
+
+        // best chromosome from the final population
+        Chromosome best = finalPopulation.getFittestChromosome();
+        System.out.println("Generation " + ga.getGenerationsEvolved() + ": " + best.toString());
+        System.out.println("Total execution time: " + (endTime - startTime) + "ms");
+    }
+    
+    private static List<Character> randomRepresentation(int length) {
+        return asList(RandomStringUtils.randomAscii(length));
+    }
+
+    private static List<Character> asList(String str) {
+        return Arrays.asList(ArrayUtils.toObject(str.toCharArray()));    
+    }
+    
+    private static Population getInitialPopulation() {
+        List<Chromosome> popList = new LinkedList<Chromosome>();
+
+        for (int i = 0; i < POPULATION_SIZE; i++) {
+            popList.add(new StringChromosome(randomRepresentation(DIMENSION)));
+        }
+        return new ElitisticListPopulation(popList, 2 * popList.size(), ELITISM_RATE);
+    }
+
+    /**
+     * String Chromosome represented by a list of characters.
+     */
+    public static class StringChromosome extends AbstractListChromosome<Character> {
+
+        public StringChromosome(List<Character> repr) {
+            super(repr);
+        }
+
+        public StringChromosome(String str) {
+            this(asList(str));
+        }
+
+        public double fitness() {
+            String target = TARGET_STRING;
+            int f = 0; // start at 0; the best fitness
+            List<Character> chromosome = getRepresentation();
+            for (int i = 0, c = target.length(); i < c; i++) {
+                // subtract the ascii difference between the target character and the chromosome character.
+                // Thus 'c' is fitter than 'd' when compared to 'a'.
+                f -= FastMath.abs(target.charAt(i) - chromosome.get(i).charValue());
+            }
+            return f;
+        }
+
+        @Override
+        protected void checkValidity(List<Character> repr) throws InvalidRepresentationException {
+            for (char c : repr) {
+                if (c < 32 || c > 126) {
+                    throw new InvalidRepresentationException(LocalizedFormats.INVALID_FIXED_LENGTH_CHROMOSOME);
+                }
+            }
+        }
+
+        public List<Character> getStringRepresentation() {
+            return getRepresentation();
+        }
+
+        @Override
+        public StringChromosome newFixedLengthChromosome(List<Character> repr) {
+            return new StringChromosome(repr);
+        }
+
+        @Override
+        public String toString() {
+            StringBuffer sb = new StringBuffer();
+            for (Character i : getRepresentation()) {
+                sb.append(i.charValue());
+            }
+            return String.format("(f=%s '%s')", getFitness(), sb.toString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format double: &#39;getFitness&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 164. Consider formatting this data with &#39;%f&#39; instead.